File: 0881-boats-to-save-people/0881-boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    let boats=0
    people.sort((a,b)=>a-b)
    let i=0
    let j=people.length-1
    while(i<j){
        if(people[i]+people[j]<=limit){
            boats+=1
            i++
            j--
        }else{
            boats+=1
            j--
        }
        if(i===j){
            boats+=1
        }
    }
    return boats
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int count=0;
        int l=0;
        int r=people.length-1;
        while(l<r){
            if(people[l]+people[r]<=limit){
                count+=1;
                l++;
                r--;
            }else{
                count++;
                r--;
            }
            if(l==r){
                count++;
            }
        }
        return count;
    }
}